spartan/releases/rough-rhino/aztec-spartan.sh

```diff
@@ -163,49 +159,89 @@ configure_environment() {
+INBOX_CONTRACT_ADDRESS=0xed179b78d5781f93eb169730d8ad1be7313123f4
+OUTBOX_CONTRACT_ADDRESS=0x1016b5aaa3270a65c315c664ecb238b6db270b64
+P2P_UDP_LISTEN_ADDR=0.0.0.0:${P2P_PORT}
+P2P_TCP_LISTEN_ADDR=0.0.0.0:${P2P_PORT}
+DATA_DIRECTORY=/var/lib/aztec
+PEER_ID_PRIVATE_KEY=${PEER_ID_PRIVATE_KEY}
+EOF
+
     # Generate docker-compose.yml
     cat > docker-compose.yml << EOF
-name: ${NAME}
 services:
     validator:
         network_mode: host
         restart: unless-stopped
-        environment:
-            - P2P_UDP_ANNOUNCE_ADDR=${IP}:${P2P_PORT}
-            - P2P_TCP_ANNOUNCE_ADDR=${IP}:${P2P_PORT}
-            - COINBASE=0xbaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
-            - VALIDATOR_DISABLED=false
-            - VALIDATOR_PRIVATE_KEY=${KEY}
-            - SEQ_PUBLISHER_PRIVATE_KEY=${KEY}
-            - L1_PRIVATE_KEY=${KEY}
-            - LOG_LEVEL=debug
-            - AZTEC_PORT=${PORT}
-            - P2P_ENABLED=true
-            - L1_CHAIN_ID=1337
-            - PROVER_REAL_PROOFS=true
-            - PXE_PROVER_ENABLED=true
-            - ETHEREUM_SLOT_DURATION=12sec
-            - AZTEC_SLOT_DURATION=36
-            - AZTEC_EPOCH_DURATION=32
-            - AZTEC_EPOCH_PROOF_CLAIM_WINDOW_IN_L2_SLOTS=13
-            - ETHEREUM_HOST=http://34.48.76.131:8545
-            - BOOTSTRAP_NODES=enr:-Jq4QO_3szmgtG2cbEdnFDIhpGAQkc1HwfNy4-M6sG9QmQbPTmp9PMOHR3xslfR23hORiU-GpA7uM9uXw49lFcnuuvYGjWF6dGVjX25ldHdvcmsBgmlkgnY0gmlwhCIwTIOJc2VjcDI1NmsxoQKQTN17XKCwjYSSwmTc-6YzCMhd3v6Ofl8TS-WunX6LCoN0Y3CCndCDdWRwgp3Q
-            - REGISTRY_CONTRACT_ADDRESS=0x5fbdb2315678afecb367f032d93f642f64180aa3
-            - GOVERNANCE_PROPOSER_CONTRACT_ADDRESS=0x9fe46736679d2d9a65f0992f2272de9f3c7fa6e0
-            - FEE_JUICE_CONTRACT_ADDRESS=0xe7f1725e7734ce288f8367e1bb143e90bb3f0512
-            - ROLLUP_CONTRACT_ADDRESS=0x2279b7a0a67db372996a5fab50d91eaa73d2ebe6
-            - REWARD_DISTRIBUTOR_CONTRACT_ADDRESS=0x5fc8d32690cc91d4c39d9d3abcbd16989f875707
-            - GOVERNANCE_CONTRACT_ADDRESS=0xcf7ed3acca5a467e9e704c703e8d87f634fb0fc9
-            - COIN_ISSUER_CONTRACT_ADDRESS=0xdc64a140aa3e981100a9beca4e685f962f0cf6c9
-            - FEE_JUICE_PORTAL_CONTRACT_ADDRESS=0x0165878a594ca255338adfa4d48449f69242eb8f
-            - INBOX_CONTRACT_ADDRESS=0xed179b78d5781f93eb169730d8ad1be7313123f4
-            - OUTBOX_CONTRACT_ADDRESS=0x1016b5aaa3270a65c315c664ecb238b6db270b64
-            - P2P_UDP_LISTEN_ADDR=0.0.0.0:${P2P_PORT}
-            - P2P_TCP_LISTEN_ADDR=0.0.0.0:${P2P_PORT}
+        env_file: .env
         image: aztecprotocol/aztec:698cd3d62680629a3f1bfc0f82604534cedbccf3-${ARCH}
-        command: start --node --archiver --sequencer
+        entrypoint: >
+            sh -c '
+
+            test -z "\$PEER_ID_PRIVATE_KEY" -a ! -f /var/lib/aztec/p2p-private-key && node /usr/src/yarn-project/aztec/dest/bin/index.js generate-p2p-private-key | head -1 | cut -d" " -f 3 | tee /var/lib/aztec/p2p-private-key || echo "Re-using existing P2P private key"
```

Comment:
Feels like we should have a `--json` option for `generate-p2p-private-key` so we can parse the output more reliably.